src/c++/perf_analyzer/genai-perf/genai_perf/llm_inputs/llm_inputs.py

```diff
@@ -156,36 +161,21 @@ def create_llm_inputs(
             input_type, dataset_name, starting_index, length, tokenizer
         )
 
-        if input_type == PromptSource.DATASET:
-            dataset = cls._get_input_dataset_from_url(
-                dataset_name, starting_index, length
-            )
-            generic_dataset_json = cls._convert_input_url_dataset_to_generic_json(
-                dataset
-            )
-        elif input_type == PromptSource.SYNTHETIC:
-            random.seed(random_seed)
-            synthetic_dataset = cls._get_input_dataset_from_synthetic(
-                tokenizer,
-                prompt_tokens_mean,
-                prompt_tokens_stddev,
-                num_of_output_prompts,
-            )
-            generic_dataset_json = (
-                cls._convert_input_synthetic_or_file_dataset_to_generic_json(
-                    synthetic_dataset
-                )
-            )
-        elif input_type == PromptSource.FILE:
-            input_filename = cast(Path, input_filename)
-            input_file_dataset = cls._get_input_dataset_from_file(input_filename)
-            generic_dataset_json = (
-                cls._convert_input_synthetic_or_file_dataset_to_generic_json(
-                    input_file_dataset
-                )
-            )
-        else:
-            raise GenAIPerfException("Input source is not recognized.")
+        random.seed(random_seed)
+
+        generic_dataset_json = cls.get_generic_dataset_json(
```

Comment:
Does this generic format match the other paths thru the code?

Reply:
It does! It simply pulls the large/branching logic for getting the dataset into its own function. I think this looks cleaner, and it'll reduce the messiness as we add the embeddings/rankings options (until we can hopefully find a way to make it even cleaner).

Reply:
fantastic.